### DISCUSSION TITLE
Reverting changes by hunks from commits

### DISCUSSION THREAD

**Discussion by MaroMetelski** [jesseduffield/lazygit#3157] [Q&A] [answered]
Is it possible to selectively revert changes (make changes opposite to a hunk without modifying the commit) from an existing commit?

I imagine something like that:
1. View file changes in a commit
2. Select a hunk
3. Run a command (?)
4. Opposite change that reverts a hunk is created
5. This change can be added and separately committed

I vaguely remember such command from Magit which was useful to selectively remove changes from a commit, it made this easier than, for example, running a `reset --soft`, and then committing again with reused message.

I see that when I navigate to file changes in a commit I'm somewhat pointed to "custom patch" mechanism but I'm not sure what that does.

**Comment by mark2185** [comment, marked as answer]
You're looking at the right thing, it is done through custom patches, [here's a snippet](https://github.com/jesseduffield/lazygit#rebase-magic-custom-patches) that demonstrates it.

**Reply by MaroMetelski** [reply]
Thanks for clarification. That's exactly what I've been looking for!
